### ISSUE TITLE
Git - Stage change with non UTF-8 file show all special characters as changes 

### ISSUE THREAD

**Issue by izacsc** [microsoft/vscode#155658]
Issue Type: <b>Bug</b>

1. Create a non UTF-8 file ( e.g. windows 1252 ) in a git repository with special characters such as á, ã, ç, etc...
2. Make changes in some lines in this file.
3. Use the UI to stage one line.
3. All 'special' characters are now marked as changes and shown as characters with encoding problems ( UTF-8 characters in a windows 1252 file ) 

VS Code version: Code 1.69.2 (3b889b090b5ad5793f524b5d1d39fda662b96a2a, 2022-07-18T16:12:52.460Z)
OS version: Windows_NT x64 10.0.18362
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.88GB (6.17GB free)|
|Process Argv|--crash-reporter-id 894f54ae-c2f3-4c75-bfc8-7f4ac5dc9e16|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (29)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|13.3.1
npm-intellisense|chr|1.4.2
vscode-eslint|dba|2.2.6
xml|Dot|2.5.1
gitlens|eam|12.1.2
vscode-npm-script|eg2|0.3.28
vscode-engpro-extension|eng|1.7.3
prettier-vscode|esb|9.5.0
remotehub|Git|0.39.2022071901
vscode-test-explorer|hbe|2.21.1
search-node-modules|jas|1.3.0
angular-file-changer|joh|0.0.4
vscode-peacock|joh|4.0.1
advpl-unit-test|Kil|0.1.0
advpl-vscode|Kil|0.14.11
dotenv|mik|1.0.1
azure-repos|ms-|0.15.2022071901
live-server|ms-|0.2.12
remote-repositories|ms-|0.17.2022071901
test-adapter-converter|ms-|0.1.6
vsliveshare|ms-|1.0.5641
vscode-jest|Ort|4.6.0
vscode-commons|red|0.0.6
vscode-yaml|red|1.9.1
angular-cli|seg|0.0.19
code-spell-checker|str|2.3.1
code-spell-checker-portuguese-brazilian|str|2.0.7
vscodeintellicode|Vis|1.2.22
JavaScriptSnippets|xab|1.8.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vscscmwlcmt:30465135
cppdebug:30492333
pylanb8912cf:30529770
vsclangdc:30486549

```

</details>

**Comment by izacsc** [comment]
![Code_qxit0nqW5q](https://user-images.githubusercontent.com/423449/179850188-4b707f17-cd93-4d69-b343-a714899197a9.gif)

**Comment by izacsc** [comment]
Any updates on this? It seems that the new merge editor also has the same problem.

**Comment by lszomoru** [comment]
@Jeje2201, could you please try VS Code Insiders (1.97) as we have made some changes in this area. Thanks!

**Comment by lszomoru** [comment]
I believe this is a duplicate of https://github.com/microsoft/vscode/issues/111915 that has been fixed in 1.97.

**Comment by Jeje2201** [comment]
I can't try the Insider build, so I'll wait for the 1.97 public release to test it. However, seeing that people who have tried 1.97 are all reporting that it fixes the issue, I’m optimistic it will resolve mine as well.

Thanks for sharing the link to the original issue!

**Comment by Jeje2201** [comment]
1.97 released, and yes, it works like a charm, thank you!

**Comment by izacsc** [comment]
It didnt work for me. Same behavior.

**Comment by izacsc** [comment]
![Image](https://github.com/user-attachments/assets/c3b9e0a6-2b96-4d25-9315-a471d6a4ee13)

![Image](https://github.com/user-attachments/assets/c198ded0-24df-4a6c-8af2-fbf207da3e0e)

**Comment by lszomoru** [comment]
@izacsc, can you please confirm that you have the `files.autoGuessEncoding` setting enabled?

**Comment by izacsc** [comment]
It is off in my config. Should It be on? Also the file is selected as windows1252, as you can see in the bottom bar. I do have custom encoding for this filetype in my config:

![Image](https://github.com/user-attachments/assets/160dacd8-4858-4427-bef3-94c27768f328)

**Comment by lszomoru** [comment]
Yes, the setting has to be on at the moment. Please turn it on and see if you can repro the issue. Thanks!

**Comment by izacsc** [comment]
With the setting enabled it works as expected! Thanks
